src/application/engine/Menu.java

```diff
@@ -1,4 +1,78 @@
 package application.engine;
 
+import application.coffe.Coffee;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
 public class Menu {
+    private final Scanner scanner;
+    private final Machine machine;
+
+    public Menu(Machine machine) {
+        this.machine = machine;
+        this.scanner = new Scanner(System.in);
+    }
+
+    public void run() {
+        String result = "";
+        boolean control = true;
+        while (control) {
+            System.out.println("Write action (buy, fill, take, remaining, exit):");
+            var response = scanner.next();
+            switch (response) {
+                case "buy" -> System.out.println(makingCoffee(scanner, machine, result));
+                case "fill" -> makeFill(scanner, machine);
+                case "take" -> System.out.println(machine.take());
+                case "remaining" -> System.out.println(machine.status());
+                default -> System.out.println("No such command");
+                case "exit" -> control = false;
+            }
+        }
+        scanner.close();
+    }
+
+    private String makingCoffee(Scanner scanner, Machine machine, String result) {
+        System.out.println("What do you want to buy? 1 - espresso, 2 - latte, 3 - cappuccino, back - to main menu:");
+        Map<String, Coffee> recipeCoffee = recipeCoffee();
+        String number = scanner.next();
+        switch (number) {
+            case "1":
```

Comment:
What do you think about adding to class Coffee a new field `name` ?
I suggest  to use command number as a key in recipeCoffee
It is better to move the initialization of this map to `main`